demos/matlabanim.py

```diff
@@ -0,0 +1,224 @@
+                          node_lat=nc['lat_psi'],
+                          edge1_lon=nc['lon_u'],
+                          edge1_lat=nc['lat_u'],
+                          edge2_lon=nc['lon_v'],
+                          edge2_lat=nc['lat_v'],
+                          )
+    sgrid.u = pysgrid.variables.SGridVariable(data=nc['u'])
+    sgrid.v = pysgrid.variables.SGridVariable(data=nc['v'])
+    sgrid.angles = pysgrid.variables.SGridVariable(data=nc['angle'])
+
+
+points = np.stack((lons, lats), axis=-1).reshape(-1, 2)
+
+ind = sgrid.locate_faces(points)
+
+u_grid = sgrid.infer_grid(sgrid.u)
+v_grid = sgrid.infer_grid(sgrid.v)
+
+u_ind = sgrid.translate_index(
+    points, ind, u_grid, slice_grid=False)
+v_ind = sgrid.translate_index(
+    points, ind, v_grid, slice_grid=False)
+
+u_alphas = sgrid.interpolation_alphas(
+    points, None, u_grid, _translated_indices=u_ind)
+v_alphas = sgrid.interpolation_alphas(
+    points, None, v_grid, _translated_indices=v_ind)
+
+ang_ind = ind + [1, 1]
+angles = sgrid.angles[:][ang_ind[:, 0], ang_ind[:, 1]]
+
+# ims is a list of lists, each row is a list of artists to draw in the
+# current frame; here we are just animating one artist, the image, in
+# each frame
+fig, ax = make_map()
+print fig
+print ax
+index = 0
+ax.coastlines('10m')
+
+t = np.linspace(0, maxslice, maxslice * fps)
+cs = qv = tl = None
+time_str = timeobj.time_str(0)
+tl = ax.text(0, 1, time_str, bbox=dict(
+    facecolor='white', alpha=0.8), transform=ax.transAxes)
+
+
+def gen_map(k):
+    global t, index, cs, qv, tl, timeobj
+    tindex = t[index]
+    if cs is not None:
+        cs.remove()
+        qv.remove()
+    time_str = timeobj.time_str(tindex)
+    tl.set_text(time_str)
+    mscale = 1
+    vscale = 15
+    scale = 0.04
+    lon_data = lons
+    lat_data = lats
+
+    print tindex
+    print time_str
+    u_rot, v_rot = interpolated_velocities(
+        sgrid, points, timeobj, tindex, sgrid.u, sgrid.v, u_alphas, v_alphas, u_ind, v_ind)
+    u_rot, v_rot = rotate_vectors(u_rot, v_rot, angles)
+    u_rot = u_rot.reshape(600, -1)
+    v_rot = v_rot.reshape(600, -1)
+
+    uv_vector_sum = vector_sum(u_rot, v_rot)
+
+    kw = dict(scale=1.0 / scale, pivot='middle', width=0.003, color='black')
+    cs = plt.pcolormesh(lon_data[::mscale, ::mscale],
+                        lat_data[::mscale, ::mscale],
+                        uv_vector_sum[::mscale, ::mscale], zorder=1, cmap=plt.cm.rainbow)
+    qv = plt.quiver(lon_data[::vscale, ::vscale], lat_data[::vscale, ::vscale],
+                    u_rot[::vscale, ::vscale], v_rot[::vscale, ::vscale], zorder=2, **kw)
+    index += 1
+    return cs, qv, tl
+
+print 'creating animation'
+ani = animation.FuncAnimation(
+    fig, gen_map, frames=maxslice * fps - 1, interval=100, blit=True, repeat=False)
+
+writer = FFMpegWriter(fps=fps, bitrate=1500)
+# plt.show()
+
+print 'saving'
+ani.save('currents_movie.mp4', writer=writer)
+print 'done'
```

Comment:
The whole stack is OK with python 3 and we are recommend python 3 to ioos users.  It would be nice to get the examples and demos py3k compat.